### DISCUSSION TITLE
Dutch translations

### DISCUSSION THREAD

**Discussion by eawmaas** [xenocrat/chyrp-lite#197] [General]
I have translated Chyrp Lite to Dutch, including the four not pre-installed mods.
I have a map with the exact structure of Chyrp Lite, but only containing the locales inside the right maps, it's a ZIP.

I tried with a pull request (i believe that's the thing to do) but it was denied, so I try it here.
See attachment.
[locales_nl_NL.zip](https://github.com/xenocrat/chyrp-lite/files/12456588/locales_nl_NL.zip)

**Comment by xenocrat** [comment]
This is fantastic - thank you for this contribution! I will add these translations to the project and credit you for them.

**Comment by eawmaas** [comment]
Hi, I am working on some revisions of the translations for Dutch, because under php 8.2 some break the PHP-code because of the apostrofes (I tested the translations on 8.0 because I could not run 8.2 due to GD not functioning at that time, which does function now).

PHP 8.2 is quite unforgivable with apostrofes and I have seen in other languages too, including English, that some functionalities don't work as expected because these files have apostrofes too and as I want to learn more about Chyrp Lite I am giving this some extra attention at the moment.

One thing that I can say for sure now is that the .pot-files **in the themes** contain the value "F Y" to be translated, but do **not** do that, it will break up some dates. This value shouldn't be translatable, it's the date-format that PHP uses and if you want that to be adjustable, it should be under settings instead of a translatable value.

So a version 2 of the Dutch translations will come and I will do some more tests on other languages too which contain the apostrofe (maybe I can use &#39; or &apos; for that or else just get rid of the apostrofes in the translations completely).

Will come back on this later!

**Comment by xenocrat** [comment]
Don’t worry about the apostrophes in translations - I noticed an issue with some of them in the JavaScript code Chyrp Lite generates and I fixed the issue, [here](https://github.com/xenocrat/chyrp-lite/commit/6a13e5254f56b14796ed66c2488504e9ae94ee09) and [here](https://github.com/xenocrat/chyrp-lite/commit/8c58b8eed38f4fcb53cec1db9915d70a5c3749c0). If you see any other occurrences just let me know and I will fix them.

I’ll look into the “F Y” issue.

**Comment by eawmaas** [comment]
Okay thanks, I will just review the Dutch translations which will be quite easy to do since I already have the .po files and in the meanwhile look for more instances of the apostrophe in the English texts.

I have seen some more issues, but have to set the translations right first before I can be sure, will come back on this with all my findings!

**Comment by eawmaas** [comment]
What did I do?

I did a search in all locale-folders (on .po and .pot files) for apostrophes, there were many and not only in the English files but most of them in Italian and some in German.

I used notepad++ to replace all instances of ' with ` in .pot and .po files (not .mo because that will corrupt the file)

After the replacements I opened all Non-English .po-files (Except Arabic, Korean and Chinese) who now contain the new ` character and saved them again which creates all new .mo files, then I uploaded it all and tested the changes, looks like everything is working perfectly with all extensions activated except the two conflicting ones.

If you want me to try the same with Arabic, Korean and Chinese please do say so, if it works it might then be that the "addslashes-function" isn't necessary anymore for apostrophes, which might improve performance a liitle.

I also translated the Readme which I forgot the last time and in the future I might translate Chyrp Lite to French and Portuguese too and complete the German translations, I speak 5 languages.



These were all the apostrophes in English .pot files (they are replaced by ` in the zipfile):

  admin\locale\en_US\LC_MESSAGES\admin.pot (2 x)
	Line 1136: msgid "The post's feather"
	Line 1144: msgid "The plural form of the post's feather"
	
  feathers\link\locale\en_US\LC_MESSAGES\link.pot (1 x)
	Line 25: msgid "URL can't be empty."
	
  feathers\photo\locale\en_US\LC_MESSAGES\photo.pot (1 x)
	Line 10: msgid "This text provides a brief description of the photo; it is useful for visually impaired visitors, and it will be displayed if the image format is not supported by the visitor's web browser or if the image has not yet downloaded."
	
  feathers\quote\locale\en_US\LC_MESSAGES\quote.pot (1 x)
	Line 18: msgid "Quote can't be empty."
	
  feathers\text\locale\en_US\LC_MESSAGES\text.pot (1 x)
	Line 21: msgid "Body can't be blank."
	
  includes\locale\en_US\LC_MESSAGES\chyrp.pot (3 x)
	Line 1054: msgid "There aren't enough posts for random selection."
	Line 1125: msgid "You aren't logged in."
	Line 1728: msgid "Delete <em>install.php</em>, you won't need it anymore."
	
  modules\comments\locale\en_US\LC_MESSAGES\comments.pot (3 x)
	Line 213: msgid "Message can't be blank."
	Line 218: msgid "Author can't be blank."
	Line 223: msgid "Email address can't be blank."
	
  modules\recaptchav2\locale\en_US\LC_MESSAGES\recaptchav2.pot (5 x)
	Line  54: msgid "Be sure to select \"reCAPTCHA v2\", and then \"I'm not a robot\" Checkbox."
	Line  78: msgid "Enabling 'Debug' allows you to view the 'response' from Google on form (registration form) submit."
	
  themes\blossom\locale\en_US\LC_MESSAGES\blossom.pot (7 x)
	Line 205: msgid "Author's website"
	Line 209: msgid "Author's email"
	Line 246: msgid "%s's comment"
	Line 381: msgid "There aren't any posts in the timeframe you specified."
	Line 390: msgid "This user hasn't created any posts."
	Line 403: msgid "You don't have any drafts."
	Line 438: msgid "There aren't any updated posts."
	
  themes\sparrow\locale\en_US\LC_MESSAGES\sparrow.pot (7 x)
	Line 105: msgid "Author's website"
	Line 109: msgid "Author's email"
	Line 146: msgid "%s's comment"
	Line 348: msgid "There aren't any posts in the timeframe you specified."
	Line 357: msgid "This user hasn't created any posts."
	Line 370: msgid "You don't have any drafts."
	Line 401: msgid "There aren't any updated posts."
	
  themes\topaz\locale\en_US\LC_MESSAGES\topaz.pot (7 x)
	Line 216: msgid "Author's website"
	Line 220: msgid "Author's email"
	Line 257: msgid "%s's comment"
	Line 392: msgid "There aren't any posts in the timeframe you specified."
	Line 401: msgid "This user hasn't created any posts."
	Line 414: msgid "You don't have any drafts."
	Line 449: msgid "There aren't any updated posts."
	
  themes\umbra\locale\en_US\LC_MESSAGES\umbra.pot (7 x)
	Line  79: msgid "Author's website"
	Line  83: msgid "Author's email"
	Line 120: msgid "%s's comment"
	Line 309: msgid "There aren't any posts in the timeframe you specified."
	Line 318: msgid "This user hasn't created any posts."
	Line 331: msgid "You don't have any drafts."
	Line 362: msgid "There aren't any updated posts."
	
  themes\virgula\locale\en_US\LC_MESSAGES\virgula.pot (6 x)
	Line  74: msgid "Author's website"
	Line  78: msgid "Author's email"
	Line 299: msgid "There aren't any posts in the timeframe you specified."
	Line 308: msgid "This user hasn't created any posts."
	Line 321: msgid "You don't have any drafts."
	Line 368: msgid "There aren't any updated posts."
	
[locales-without-apostrophes.zip](https://github.com/xenocrat/chyrp-lite/files/12486666/locales-without-apostrophes.zip)

**Comment by xenocrat** [comment]
Hello there,

> I used notepad++ to replace all instances of ' with ` in .pot and .po files

This definitely isn't needed - I have these translations functioning without issues and so do other users. Let's take a step back and try to debug the problems you are experiencing. I don't want to grammatically mangle text to work around a problem that can be fixed in a more elegant way.

My advice is for you to download the ["develop"](https://github.com/xenocrat/chyrp-lite/tree/develop) branch of Chyrp Lite and upgrade your installation to this; it contains all the most recent changes I have made, including the addition of your translations and the fixes I made in response to the problems your translations revealed in the codebase. The next thing to do is enable [DEBUG mode](https://github.com/xenocrat/chyrp-lite/wiki/Debug-and-Tester-Modes), which will put Chyrp Lite in a stricter mode and report more information when problems occur.

> it might then be that the "addslashes-function" isn't necessary anymore for apostrophes, which might improve performance a liitle.

PHP's `addslashes()` function is very fast so we don't need to worry about using it whenever it is the right solution. Proper "escaping" and entity-encoding is always the right way to solve a problem like this - that's the reason you'll see the [`fix()`](https://github.com/xenocrat/chyrp-lite/blob/master/includes/helpers.php#L1419) helper function used in many, many places throughout the codebase.

> One thing that I can say for sure now is that the .pot-files in the themes contain the value "F Y" to be translated, but do not do that, it will break up some dates.

"F Y" is a time-formatting code. This can be translated but it usually won't be necessary, because Chyrp Lite tries to use PHP functions that already output the text in the correct language. If you do want to change the way time is displayed, perhaps because the English format I've chosen looks weird in Dutch, you must use the [the formatting codes understood by PHP's `date()` function](https://www.php.net/manual/en/datetime.format.php). For example, "F Y" means "A full textual representation of a month" followed by "A full numeric representation of a year". You might decide that "m.Y" (08.2023) is more commonly used in Dutch and looks more correct to Dutch speakers.

> I also translated the Readme which I forgot the last time and in the future I might translate Chyrp Lite to French and Portuguese too and complete the German translations, I speak 5 languages.

Fantastic, thank you! A complete German translation has been on my wish list for some time - and all additional languages are gratefully received.

**Reply by eawmaas** [reply]
Okay, I'll get the develop-branch and explore that, wasn't even aware of it's existance being a github-newbie, so thanks for the directions.

I just love the program and want to learn more about it, I might have a few issues, when I am sure it's not related to my installation or environment i'll put them under issues.

Good luck with this beautiful piece of software1
